src/main/java/uk/gov/companieshouse/acsp/manage/users/service/AcspMembersService.java

```diff
@@ -44,4 +53,27 @@ public List<AcspMembership> fetchAcspMemberships(final User user, final boolean
 
     return acspMembershipListMapper.daoToDto(acspMembers, user);
   }
+
+    @Transactional( readOnly = true )
+    public AcspMembers fetchAcspMembers( final AcspDataDao acspData, final boolean includeRemoved, final String userId, final String role, final int pageIndex, final int itemsPerPage ) {
+        final var acspNumber = acspData.getId();
+
+        if ( Objects.isNull( acspNumber ) ){
+            throw new NullPointerException( "acspNumber is null." );
```

Comment:
Illegal argument is probably more appropriate, as a null pointer hasn't actually happened yet, and it's more reflective of a missing required thing:

```suggestion
            throw new IllegalArgumentException( "acspNumber is null." );
```

Reply:
Changed it to IllegalArgumentException